swap/src/monero/wallet.rs

```diff
@@ -117,46 +116,45 @@ impl WatchForTransfer for Wallet {
         let wallet = self.inner.clone();
 
         let confirmations = Arc::new(AtomicU32::new(0u32));
-        let res = (move || {
+
```

Comment:
I got this compiling but I am not sure if I changed the behaviour

Reply:
Looks identical to me.

Reply:
If this compiles, you can likely remove all the clones in the beginning now.